### PR TITLE
Add test for contract

### PR DESCRIPTION
**Pull request by GNaD13**
#660 

add test case: Can a contract close a wasm ibc channel that is not owned by the contract?

### CODE REVIEW

File: x/wasm/keeper/keeper.go
Comment (jhernandezb) [inline x/wasm/keeper/keeper.go:963]: I don't think we should expose this, or at least expose it for tests only.

Comment (GNaD13) [reply x/wasm/keeper/keeper.go:963]: tysm, I'll fix it

Comment (GNaD13) [reply x/wasm/keeper/keeper.go]: Hi @jhernandezb , I have fixed this problem. Tysm for your reviews. I've learned a lot <3

**Review by ethanfrey — COMMENTED** [review]
Looks like a useful test case. Thank you for the contribution.

I would wait for @alpe to review it for code quality before merge, but he is off sick this week.

**Comment by alpe** [comment]
I will 👁️ at this tomorrow when I feel better. 🦷 OP.

File: x/wasm/relay_test.go
Comment (alpe) [inline x/wasm/relay_test.go:562]: Nit: it is not capturing anything. How about:
```suggestion
type closeChannelContract struct {
```

File: x/wasm/relay_test.go
Comment (alpe) [inline x/wasm/relay_test.go:582]: very nice!

File: x/wasm/relay_test.go
Comment (alpe) [inline x/wasm/relay_test.go:585]: nit: uppercase IBC
```suggestion
type closeIBCChannel struct {
```

File: x/wasm/relay_test.go
Comment (alpe) [inline x/wasm/relay_test.go:449]: Good setup!

File: x/wasm/ibctesting/chain.go
Comment (alpe) [inline x/wasm/ibctesting/chain.go:289]: SignAndDeliver is a monster :-)

**Review by alpe — APPROVED** [review]
Thanks for providing this test case. 🌷 
The IBC mock tests are super straight forward but you did this very well! I have marked some minor nits, if you want to polish your code. But otherwise this is great! Happy to merge it.

File: x/wasm/relay_test.go
Comment (GNaD13) [reply x/wasm/relay_test.go]: I've changed this

File: x/wasm/relay_test.go
Comment (GNaD13) [reply x/wasm/relay_test.go:582]: Thank you!!

File: x/wasm/relay_test.go
Comment (GNaD13) [reply x/wasm/relay_test.go]: I've changed this

**Comment by GNaD13** [comment]
> Thanks for providing this test case. tulip The IBC mock tests are not super straight forward but you did this very well! I have marked some minor nits, if you want to polish your code. But otherwise this is great! Happy to merge it.

tysm for your review :3

**Pull request merged**
